internal/api/middleware/rbac.go

```diff
@@ -0,0 +1,96 @@
+package middleware
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"io"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+	"github.com/redhatinsights/ros-ocp-backend/internal/config"
+	"github.com/redhatinsights/ros-ocp-backend/internal/logging"
+	"github.com/redhatinsights/ros-ocp-backend/internal/types"
+	"github.com/sirupsen/logrus"
+)
+
+var cfg *config.Config = config.GetConfig()
+var log *logrus.Logger = logging.GetLogger()
+
+func Rbac(next echo.HandlerFunc) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		permissions := get_user_permissions(c.Request().Header.Get("X-Rh-Identity"))
```

Comment:
Two methods with same naming i.e `get_user_permissions` and different set of arguments (one accepts context whereas other accepts identity header) in two files. Could you please rename those to avoid confusion or make it single method to address both requirements?
Thoughts?

Reply:
Done. Changed the function name to `get_user_permissions_from_rbac`.